### PR TITLE
Remove unused and uneccessary JSONExceptions

### PR DESCRIPTION
**Pull request by rachael-oregan**
## Motivation

Issue #98 

## Description

Refactored throws `JSONException` to `try/catch` where functions are not parsing JSON & remove unused `JSONException` from functions

### CODE REVIEW

File: auth/src/main/java/org/aerogear/mobile/auth/utils/UserIdentityParser.java [hunk 1]
```diff
@@ -56,13 +61,15 @@ public UserIdentityParser(final OIDCCredentials credential,
...
         String firstName = "";
         if (userIdentity != null && userIdentity.has(FIRST_NAME)) {
-            firstName = userIdentity.getString(FIRST_NAME);
+            try {
+                firstName = userIdentity.getString(FIRST_NAME);
+            } catch (JSONException e) {
+                LOG.debug(TAG, "Failed to get user first name from user identity", e);
+            }
         }
         return firstName;
```
Comment (wei-lee) [inline auth/src/main/java/org/aerogear/mobile/auth/utils/UserIdentityParser.java]: This function can be replaced with a one liner:

```java
return userIdentity == null ? "" : userIdentity.optString(FIRST_NAME, "");
```

File: auth/src/main/java/org/aerogear/mobile/auth/utils/UserIdentityParser.java [hunk 2]
```diff
@@ -71,13 +78,15 @@ public String parseFirstName() throws JSONException {
...
         String lastName = "";
         if (userIdentity != null && userIdentity.has(LAST_NAME)) {
-            lastName = userIdentity.getString(LAST_NAME);
+            try {
+                lastName = userIdentity.getString(LAST_NAME);
+            } catch (JSONException e) {
+                LOG.debug(TAG, "Failed to get user last name from user identity", e);
+            }
         }
         return lastName;
```
Comment (wei-lee) [inline auth/src/main/java/org/aerogear/mobile/auth/utils/UserIdentityParser.java]: same as above

File: auth/src/main/java/org/aerogear/mobile/auth/utils/UserIdentityParser.java (hunk 1 above)
Comment (rachael-oregan) [reply auth/src/main/java/org/aerogear/mobile/auth/utils/UserIdentityParser.java]: Nice, I'll refactor these functions

**Comment by rachael-oregan** [comment]
@wei-lee @ziccardi mind having another look when ye have some time. Thanks

**Comment by rachael-oregan** [comment]
@ziccardi there are not being used externally at the moment no. I think it makes sense to make these private actually you're right

**Comment by rachael-oregan** [comment]
@ziccardi mind having another look please

**Pull request merged**